SumoLogic.Logging.NLog/BufferedSumoLogicTarget.cs

```diff
@@ -319,19 +345,13 @@ protected override void Write(LogEventInfo logEvent)
                 return;
             }
 
-            var bodyBuilder = new StringBuilder();
-            using (var textWriter = new StringWriter(bodyBuilder, CultureInfo.InvariantCulture))
+            var body = this.Layout?.Render(logEvent) ?? string.Empty;
+            if (body.Length < Environment.NewLine.Length || body[body.Length - 1] != Environment.NewLine[Environment.NewLine.Length - 1])
             {
-                textWriter.Write(Layout.Render(logEvent));
-                if (logEvent.Exception != null && this.AppendException)
-                {
-                    textWriter.Write(logEvent.Exception.ToString());
-                }
-
-                textWriter.WriteLine();
+                body = string.Concat(body, Environment.NewLine);

```

Comment:
The exception appending feature is removed?

Reply:
Just like log4net and Serilog, then it is better to let the logging framework handling the formatting, Sadly enough it is hard to remove this hack without breaking changes. If you look at the code in InitializeTarget then you see the Layout is "fixed" depending on having set **AppendException**. I would prefer a breaking change where **AppendException** was made obsolete AND not referenced.

Reply:
I'm ok with this kind of breaking change, but you should update the doc to describe it. people will not look into the PR to find the answer, they will simply think it's a bug

Reply:
Have now updated the documentation about **AppendException** is now obsolete.